### PR TITLE
backport: Merge bitcoin#21818, 21752

### PR DESCRIPTION
**Pull request by vijaydasmp**
bitcoin backports

### CODE REVIEW

File: src/policy/policy.cpp [hunk 1]
```diff
@@ -21,7 +21,7 @@ CAmount GetDustThreshold(const CTxOut& txout, const CFeeRate& dustRelayFeeIn)
     // need a CTxIn of at least 148 bytes to spend:
     // so dust is a spendable txout less than
     // 182*dustRelayFee/1000 (in satoshis).
-    // 546 satoshis at the default rate of 3000 sat/kB.
+    // 546 satoshis at the default rate of 3000 sat/kvB.
```
Comment (knst) [inline src/policy/policy.cpp:24]: in our case we don't have segregated witness txes; size of tx in bytes are equal to virtual size.

I think this backport 21752 is DNM

File: src/init.cpp [hunk 2]
```diff
@@ -732,7 +732,7 @@ void SetupServerArgs(NodeContext& node)
     argsman.AddArg("-maxtipage=<n>", strprintf("Maximum tip age in seconds to consider node in initial block download (default: %u)", DEFAULT_MAX_TIP_AGE), ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::DEBUG_TEST);
     argsman.AddArg("-mocktime=<n>", "Replace actual time with " + UNIX_EPOCH_TIME + "(default: 0)", ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::DEBUG_TEST);
     argsman.AddArg("-minsporkkeys=<n>", "Overrides minimum spork signers to change spork value. Only useful for regtest and devnet. Using this on mainnet or testnet will ban you.", ArgsManager::ALLOW_ANY, OptionsCategory::DEBUG_TEST);
-    argsman.AddArg("-printpriority", strprintf("Log transaction fee per kB when mining blocks (default: %u)", DEFAULT_PRINTPRIORITY), ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::DEBUG_TEST);
+    argsman.AddArg("-printpriority", strprintf("Log transaction fee rate in " + CURRENCY_UNIT + "/kvB when mining blocks (default: %u)", DEFAULT_PRINTPRIORITY), ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::DEBUG_TEST);
```
Comment (knst) [inline src/init.cpp:735]: There's still several useful changes: "rate in " + CURRENCY_UNIT in src/init.cpp (here)
similar changes in wallet/init.cpp and wallet/rpcwallet.cpp -- maybe something else.

but `kvB` and mentioning virtual size to DNM

Comment (vijaydasmp) [reply src/init.cpp:735]: ok , keeping the useful changes and , 
removing kvB

File: src/policy/policy.cpp (hunk 1 above)
Comment (vijaydasmp) [reply src/policy/policy.cpp:24]: reverted this change

File: doc/release-notes-18335.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+RPC changes
+-----------
+- The RPC server can process a limited number of simultaneous RPC requests.
```
Comment (knst) [inline doc/release-notes-18335.md:3]: DNM - we already release 18335 in previous version; release notes for it is late to provide

Comment (vijaydasmp) [reply doc/release-notes-18335.md]: Dropped it

File: src/net_processing.cpp [hunk 4]
```diff
@@ -3020,8 +3020,8 @@ void PeerManagerImpl::ProcessMessage(
             // empty and no one will know who we are, so these mechanisms are
             // important to help us connect to the network.
             //
-            // We skip this for BLOCK_RELAY peers to avoid potentially leaking
-            // information about our BLOCK_RELAY connections via address relay.
+            // We skip this for block-relay-only peers to avoid potentially leaking
+            // information about our block-relay-only connections via address relay.
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply src/net_processing.cpp:3024]: Should probably drop instead of partial

[the remark this answers is not in the document]
Comment (vijaydasmp) [reply src/net_processing.cpp:3024]: sure, will drop it from this PR and do it in full in another PR. As [bitcoin#19858](https://github.com/bitcoin/bitcoin/pull/19858) is merged now through https://github.com/dashpay/dash/pull/5869

**Review by knst — APPROVED** [review]
utACK

why is it draft? @vijaydasmp

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**